ext/handlers/filters/choseninlineresult.go

```diff
@@ -0,0 +1,51 @@
+package filters
+
+import (
+	"strings"
+
+	"github.com/PaulSonOfLars/gotgbot/v2"
+)
+
+type ChosenInlineResult func(cir *gotgbot.ChosenInlineResult) bool
+
+func (f ChosenInlineResult) And(f2 ChosenInlineResult) ChosenInlineResult {
+	return func(cir *gotgbot.ChosenInlineResult) bool {
+		return f(cir) && f2(cir)
+	}
+}
+
+func (f ChosenInlineResult) Or(f2 ChosenInlineResult) ChosenInlineResult {
+	return func(cir *gotgbot.ChosenInlineResult) bool {
+		return f(cir) || f2(cir)
+	}
+}
+
+func (f ChosenInlineResult) Not() ChosenInlineResult {
+	return func(cir *gotgbot.ChosenInlineResult) bool {
+		return !f(cir)
+	}
+}
+
+func ChosenResultUserID(id int64) ChosenInlineResult {
+	return func(cir *gotgbot.ChosenInlineResult) bool {
+		return cir.From.Id == id
+	}
+}
+
+func ChosenResultQuery(q string) ChosenInlineResult {
+	return func(cir *gotgbot.ChosenInlineResult) bool {
+		return cir.Query == q
+	}
+}
+
+func ChosenResultPrefix(prefix string) ChosenInlineResult {
```

Comment:
Would it be better to call this `ChosenResultQueryPrefix` instead? Would make it easier to read the code and know what the prefix is for. What do you think?

Reply:
(of course, i see this isnt the case for the current callback query filter; If you agree with the naming scheme, I'll rename that in a later commit)

Reply:
Done!